tests/e2e/e2e_test.go

```diff
@@ -358,3 +374,370 @@ func (s *IntegrationTestSuite) verifyChainPassesUpgradeHeight(c *chain, valIdx,
+6. gov propose change back to initial globalfee = 0.00001photon, This is for not influence other e2e tests.
+*/
+func (s *IntegrationTestSuite) TestGlobalFees() {
+	chainAAPIEndpoint := fmt.Sprintf("http://%s", s.valResources[s.chainA.id][0].GetHostPort("1317/tcp"))
+
+	submitterAddr, err := s.chainA.validators[0].keyInfo.GetAddress()
+	s.Require().NoError(err)
+	submitter := submitterAddr.String()
+	recipientAddress, err := s.chainA.validators[1].keyInfo.GetAddress()
+	s.Require().NoError(err)
+	recipient := recipientAddress.String()
+	var beforeRecipientPhotonBalance sdk.Coin
+	s.Require().Eventually(
+		func() bool {
+			beforeRecipientPhotonBalance, err = getSpecificBalance(chainAAPIEndpoint, recipient, photonDenom)
+			s.Require().NoError(err)
+
+			return beforeRecipientPhotonBalance.IsValid()
+		},
+		10*time.Second,
+		5*time.Second,
+	)
+	if beforeRecipientPhotonBalance.Equal(sdk.Coin{}) {
+		beforeRecipientPhotonBalance = sdk.NewCoin(photonDenom, sdk.ZeroInt())
+	}
+
+	sendAmt := int64(1000)
+	token := sdk.NewInt64Coin(photonDenom, sendAmt) // send 1000photon each time
+	sucessBankSendCount := 0
+	// ---------------------------- test1: globalfee empty --------------------------------------------
+	// prepare gov globalfee proposal
+	emptyGlobalFee := sdk.DecCoins{}
+	s.writeGovParamChangeProposalGlobalFees(s.chainA, emptyGlobalFee)
+
+	// gov proposing new fees
+	proposalCounter++
+	s.T().Logf("Proposal number: %d", proposalCounter)
+	s.T().Logf("Submitting, deposit and vote legacy Gov Proposal: change global fees empty")
+	s.submitLegacyGovProposal(chainAAPIEndpoint, submitter, fees.String(), "param-change", proposalCounter, "/root/.gaia/config/proposal_globalfee.json")
+	s.depositGovProposal(chainAAPIEndpoint, submitter, fees.String(), proposalCounter)
+	s.voteGovProposal(chainAAPIEndpoint, submitter, fees.String(), proposalCounter, "yes", false)
+
+	// query the proposal status and new fee
+	s.Require().Eventually(
+		func() bool {
+			proposal, err := queryGovProposal(chainAAPIEndpoint, proposalCounter)
+			s.Require().NoError(err)
+			return (proposal.GetProposal().Status == govv1beta1.StatusPassed)
+		},
+		15*time.Second,
+		5*time.Second,
+	)
+	var globalFees sdk.DecCoins
+	s.Require().Eventually(
+		func() bool {
+			globalFees, err = queryGlobalFees(chainAAPIEndpoint)
+			s.T().Logf("After gov new global fee proposal: %s", globalFees.String())
+			s.Require().NoError(err)
+
+			// attention: when global fee is empty, when query, it shows empty rather than default ante.DefaultZeroGlobalFee() = 0uatom.
+			return globalFees.IsEqual(emptyGlobalFee)
+		},
+		15*time.Second,
+		5*time.Second,
+	)
+
+	paidFeeAmt := sdk.MustNewDecFromStr(minGasPrice).Mul(sdk.NewDec(gas)).String()
+
+	s.T().Logf("test case: empty global fee, globalfee=%s, min_gas_price=%s", globalFees.String(), minGasPrice+uatomDenom)
+	s.T().Logf("Tx fee is zero coin with correct denom: uatom, fail")
```

Comment:
if the global fee is empty shouldn't the default be 0uatom, which means that a zero coin fee with uatom should pass?

Reply:
ah i see min_gas_price is not zero